onnxscript/rewriter/llama_rule_sets_tests.py

```diff
@@ -0,0 +1,128 @@
+                raise AssertionError(f"Not implemented for input {i}")
+        return feeds
+
+    def _check_model(
+        self,
+        model: onnx.ModelProto,
+        optimized_model: onnx.ModelProto,
+        feeds: dict[str, Any] | None = None,
+        atol: float = 0.0,
+        rtol: float = 1e-7,
+    ):
+        if not feeds:
+            feeds = self._get_random_inputs(model)
+        ref = onnx.reference.ReferenceEvaluator(model)
+        opt = onnx.reference.ReferenceEvaluator(optimized_model)
+        expected = ref.run(None, feeds)
+        got = opt.run(None, feeds)
+        self.assertEqual(len(expected), len(got))
+        for a, b in zip(expected, got):
+            np.testing.assert_allclose(a, b, atol=atol, rtol=rtol)
+
+    def _identity_models(self):
+        models = [
+            onnx.helper.make_model(
+                onnx.helper.make_graph(
+                    [
+                        onnx.helper.make_node("Transpose", ["X"], ["Y"], perm=[0, 1, 2]),
+                    ],
+                    "name",
+                    [onnx.helper.make_tensor_value_info("X", FLOAT, [None, None, None])],
+                    [onnx.helper.make_tensor_value_info("Y", FLOAT, [None, None, None])],
+                ),
+                opset_imports=[onnx.helper.make_opsetid("", 18)],
+            ),
+            onnx.helper.make_model(
+                onnx.helper.make_graph(
+                    [
+                        onnx.helper.make_node("Mul", ["X", "one"], ["Y"]),
+                    ],
+                    "name",
+                    [onnx.helper.make_tensor_value_info("X", FLOAT, [None])],
+                    [onnx.helper.make_tensor_value_info("Y", FLOAT, [None])],
+                    [
+                        onnx.numpy_helper.from_array(
+                            np.array([1], dtype=np.float32), name="one"
+                        )
+                    ],
+                ),
+                opset_imports=[onnx.helper.make_opsetid("", 18)],
+            ),
+            onnx.helper.make_model(
+                onnx.helper.make_graph(
+                    [
+                        onnx.helper.make_node("Transpose", ["X"], ["xt"], perm=[1, 0]),
+                        onnx.helper.make_node("Transpose", ["xt"], ["Y"], perm=[1, 0]),
+                    ],
+                    "name",
+                    [onnx.helper.make_tensor_value_info("X", FLOAT, [None, None])],
+                    [onnx.helper.make_tensor_value_info("Y", FLOAT, [None, None])],
+                ),
+                opset_imports=[onnx.helper.make_opsetid("", 18)],
+            ),
+        ]
+        return models
+
+    def test_llama_p0_rule_set_identity(self):
+        for model in self._identity_models():
+            ir_model = ir.serde.deserialize_model(model)
+            rule_set = llama_rule_sets.llama_p0_rule_set()
+            rule_set.apply_to_model(ir_model)
+            rewritten_model = ir.serde.serialize_model(ir_model)
```

Comment:
Prefer `model_proto` for proto messages, `model` for ir.Model 
- https://github.com/microsoft/onnxscript/issues/1381

Reply:
done